src/include/detail/linalg/tdb_matrix.h

```diff
@@ -54,9 +54,13 @@
  * it is sufficient to simply have one Matrix class and have a factory that
  * creates them by reading from TileDB.
  */
-template <class T, class LayoutPolicy = stdx::layout_right, class I = size_t>
-class tdbBlockedMatrix : public Matrix<T, LayoutPolicy, I> {
-  using Base = Matrix<T, LayoutPolicy, I>;
+template <
```

Comment:
Hmm.  For the same reasons that `Matrix` should be standalone and independent of `MatrixWithIds`, `tdbMatrix` and `tdbMatrixWithIds` should be independent of each other.

Reply:
I think we should discuss this a bit more, as this will lead to a lot of duplication of code, which has a real cost, i.e. more maintenance and harder to change things. By using this approach we only need to write the additional IDs loading and storage.

Reply:
Talked offline and have a plan around this now. Will mark this as resolved and let's discuss any other suggestions in a follow-up comment.